File: src/main/resources/output/IFC4X3_js/Entities/IfcExternalReference/IfcClassificationReference/IfcClassificationReference.js
class IfcClassificationReference extends IfcExternalReference {
    constructor() {
        /** @type {IFCCLASSIFICATIONREFERENCESELECT} */
        this.ReferencedSource = null;
        /** @type {IFCTEXT} */
        this.Description = null;
        /** @type {IFCIDENTIFIER} */
        this.Sort = null;
    }

    // === INVERSE CLAUSES ===
    // ClassificationRefForObjects : SET [0:?] OF IfcRelAssociatesClassification FOR RelatingClassification
    // HasReferences : SET [0:?] OF IfcClassificationReference FOR ReferencedSource
}

File: src/main/resources/output/IFC4X3_java/Entities/IfcExternalReference/IfcClassificationReference/IfcClassificationReference.java
public class IfcClassificationReference extends IfcExternalReference {
    public IfcClassificationReferenceSelect ReferencedSource;
    public IfcText Description;
    public IfcIdentifier Sort;

    // === INVERSE CLAUSES ===
    // ClassificationRefForObjects : SET [0:?] OF IfcRelAssociatesClassification FOR RelatingClassification
    // HasReferences : SET [0:?] OF IfcClassificationReference FOR ReferencedSource
}
